jargons/api.md

```diff
@@ -0,0 +1,7 @@
+---
+title: API
+tags:
+- programlama
+---
+
+Uygulama Programlama Arayüzü. Diğer programlarla bilgi alışverişinde bulunan programları yazan programcılar tarafından kullanılır.
```

Comment:
Anlatım sanki biraz karmaşık olmuş. Daha anlaşılır bir şekilde anlatabiliriz diye düşünüyorum.

**Benim fikrim**: API (Application Programming Interface), diğer uygulamalarla bilgi alışverişi yapılabilmesini sağlar. API'ye bir istek yapıldıktan sonra geriye bir data döner veya gönderilen datalar üzerinde genellikle sunucu taraflı işlemler yapar.

Reply:
Merhaba,

Öncelikle, API neyin kısaltması bunu eklemek faydalı olacaktır. Bu noktada @erkansivas35 ile aynı fikirdeyim. Sonrası için pek de aynı fikirde olduğumu söyleyemem. Zira, API her zaman geriye data dönecektir diye bir kaide yok. Diğer yandan, gönderilen verinin işlenip işlenmeyeceğine dair de bir kesinlik mevcut değil.

En basit anlatımla, API, bir uygulamanın başka bir uygulamayla konuşması için gerekli olan arabirimdir. Sanırım burada kafa karıştıran en büyük parça "application" tarafı. Adı geçen uygulama bir servis de olabilir, ufacık ve tek bir iş yapan bir uygulama da olabilir, koca bir işletim sisteminin kritik bir parçası da olabilir. En temelde taşıması gereken yegane özellik diğer uygulamaların kendisiyle haberleşebilmesine olanak sağlaması.

Sonuç olarak, şu ya da bu şekilde değiştirin demek haddime değil; fakat API gibi artık günlük kullanımda bile yer etmiş bir şeyi tanımlarken biraz daha basit ve genele hitap edecek şekilde yazmak gerektiğini düşünüyorum. 

--- 

Kişisel algılanmaması gereken bir not:
> "Uygulama Programlama Arayüzü. Diğer programlarla bilgi alışverişinde bulunan programları yazan programcılar tarafından kullanılır."

Hayır, illa başka programlarla bilgi alışverişinde bulunan programları yazan programcılar tarafından kullanılmak zorunda değil. Uygulamalar da birbirleriyle arada bir yazılımcı gereksinimi olmadan API aracılığıyla haberleşebilir ve bilgi alışverişinde bulunabilir.
Diğer yandan, program dediğimiz şey "a set of instructions" yani bir dizi talimattan meydana gelen ve bilgisayara ne yapması gerektiğini söyleyen yapılardır. Uygulama ise bir veya birkaç programdan oluşan ve çoğunlukla son kullancının kullanımına sunulan yapılardır. Buradan hareketle, tüm uygulamaların öyle ya da böyle birer program barındırdığını; fakat tüm programların birer uygulama olmadığını söyleyebiliriz. Application Programming Interface'teki Application kısmı bir programa değil uygulamaya karşılık geliyor. Çoğu zaman da programcılar tarafından değil yazılım geliştiriciler tarafından kullanılıyor. Madem ki "jargon" yazıyoruz, bunlardan bahsetmemek olmaz diye düşündüm.

Reply:
Rehber niteliğinde bir yanıt olmuş gayet açıklayıcı ve bilgilendirici teşekkürler.